packages/react/src/text.tsx

```diff
@@ -2,26 +2,74 @@ import { Utility } from "@hpcc-js/common";
 import * as React from "@hpcc-js/preact-shim";
 import { Rectangle } from "./shape";
 
-interface Text {
+interface TextLine {
+    text: string;
+    height?: number;
     anchor?: string;
     baseline?: string;
-    height?: number;
+    fontFamily?: string;
     fill?: string;
-    text: string;
+    y?: number;
 }
 
-export const Text: React.FunctionComponent<Text> = ({
+export const TextLine: React.FunctionComponent<TextLine> = ({
+    text,
+    height = 12,
     anchor = "middle",
     baseline = "middle",
-    height = 12,
+    fontFamily = "Verdana",
     fill = "black",
-    text = ""
-}) => <text
-        fill={fill}
+    y
+}) => {
+    return <text
+        font-family={fontFamily}
         font-size={`${height}px`}
-        dominant-baseline={baseline}
         text-anchor={anchor}
+        dominant-baseline={baseline}
+        fill={fill}
+        y={y}
     >{text}</text>;
+};
+
+interface Text {
+    text: string;
+    height?: number;
+    fontFamily?: string;
+    fill?: string;
+    onSizeUpdate?: (size: { width: number, height: number }) => void;
+}
+
+export const Text: React.FunctionComponent<Text> = ({
+    text,
+    height = 12,
+    fontFamily = "Verdana",
+    fill = "black",
+    onSizeUpdate = (size: Utility.TextSize) => { }
+}) => {
+    const [totalWidth, onTotalWidthUpdate] = React.useState(0);
+    const [totalHeight, onTotalHeightUpdate] = React.useState(0);
+    React.useEffect(() => onSizeUpdate({ width: totalWidth, height: totalHeight }), [totalWidth, totalHeight]);
+
+    const parts = text.split("\n");
+    const ts = Utility.textSize(parts);
+    onTotalWidthUpdate(ts.width);
+    onTotalHeightUpdate(parts.length * (height + 2) - 2);
+
+    const yOffset = -(totalHeight / 2) + height / 2;
+    const TextLines = parts.map((p, i) => {
+        return <g transform={`translate(0 ${0})`}>
+            <TextLine
+                text={p}
+                height={height}
+                fontFamily={fontFamily}
+                fill={fill}
+                y={yOffset + i * (height + 2)}
+            />
+        </g>;
+    });
+
+    return <>{TextLines}</>;
+};
```

Comment:
This seems like `Text` is trying to do too much. The ability to line wrap text isn't always needed. Seems like this functionality could be added by leaving `Text` alone and doing this:
```
export const WrappingText: React.FunctionComponent<IWrappingText> = ({
...
}) => {
    const TextLines = parts.map((p, i) => {
        return <g transform={`translate(0 ${yOffset + i * (height + 2)})`}>
            <Text
                text={p}
                height={height}
                fontFamily={fontFamily}
                fill={fill}
            />
        </g>;
    });
    return <g>{TextLines}</g>;
};
```

Reply:
Is this just a naming thing?  I have "TextLine" + "Text" - you have "Text" + "WrappingText"?

Reply:
I have removed "y" from Text and moved that logic into the <g> element as per your suggestion above.